spellings/action.yml

```diff
@@ -2,28 +2,102 @@ name: 'spellings'
-          do dir=${lexfile%/lexicon.txt}
-            echo $dir
-            find-unknown-comment-words --directory $dir
-            if [ $? -ne "0" ]
-            then
-              exit 1
-            fi
-          done
-        shell: bash
+    - env:
+        stepName: Install Spell
+        bashPass: \033[32;1mPASSED -
+        bashInfo: \033[33;1mINFO -
+        bashFail: \033[31;1mFAILED -
+        bashEnd:  \033[0m
+      name: ${{ env.stepName }}
+      id: spell-checker-setup
+      shell: bash
+      run: |
+        #  ${{ env.stepName }}
+        echo "::group::${{ env.stepName }}"
+        sudo apt-get install spell -y
+        sudo apt-get install util-linux -y
+        sudo apt-get install fd-find -y
+
+        # This is a 1000+ line log for the install
+        # Wrap it in an echo group
+        echo "::group::NPM Install"
+        sudo apt-get install npm -y
+        sudo npm install -g cspell
+        echo "::endgroup::"
+
+        echo "$GITHUB_ACTION_PATH" >> $GITHUB_PATH
+        export PATH="$PATH:$GITHUB_ACTION_PATH"
+
+        # cp has an error code if it's asked to copy to the same file
+        # Wrap it in a set +e in case the input path is the working directory
+        set +e
+        cp $GITHUB_ACTION_PATH/cspell.config.yaml ${{ inputs.path }}
+        set -e
+        echo "::endgroup::"
+
+        # Make sure we have all the commands we need.
+        echo -e "${{ env.bashInfo }} which getFiles ${{ env.bashEnd }}"
+        echo -e "${{ env.bashInfo }} fdfind --version ${{ env.bashEnd }}"
+        echo -e "${{ env.bashInfo }} cspell --version ${{ env.bashEnd }}"
+
+        # Only get to here if everything above passes
+        echo -e "${{ env.bashPass }} ${{ env.stepName }} ${{ env.bashEnd }}"
+
+    - env:
+        bashPass: \033[32;1mPASSED -
+        bashInfo: \033[33;1mINFO -
+        bashFail: \033[31;1mFAILED -
+        bashEnd:  \033[0m
+        stepName: Check Spelling
+      name: ${{ env.stepName}}
+      id: run-spell-checker
+      working-directory: ${{ inputs.path }}
+      shell: bash
+      run: |
+        #  ${{ env.stepName }}
+        echo "::group::${{ env.stepName }}"
+
+        # Add helper script to path
+        export PATH="$PATH:$GITHUB_ACTION_PATH"
+
+        # Get all files
+        files=$(getFiles --exclude-dirs="${{ inputs.exclude-dirs }}" --exclude-files="${{ inputs.exclude-files }}" --include-extensions="${{ inputs.include-extensions}}")
+
+        # Wrap the check in set +e so it runs against all files
+        set +e
+        exitStatus=0
+        for file in ${files[@]}; do
+          cspell --language-id C --color -c cspell.config.yaml --show-suggestions --show-context "$file"
+          if ! [ $? -eq 0 ]; then
+            exitStatus=1
+          fi
+        done
+        set -e
+
+        echo "::endgroup::"
+        if [ $exitStatus -eq 0 ]; then
+          echo -e "${{ env.bashPass }} ${{ env.stepName }} ${{ env.bashEnd }}"
```

Comment:
nit-pick: Perhaps this type of output would be improved if included in a job-summary markdown: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary

Colorized output isn't very portable, so I would generally prefer not to include it in console logs.

Reply:
I had seen issues trying to access things like that from inside of the action, as I think you have to manually pass in your github environment to them to be able to access these?
But think that's an optimization that could be added after the initial merge of this